cmake/setup_custom_targets.cmake

```diff
@@ -60,13 +60,19 @@ FOREACH(_component compat_files documentation examples parameter_gui)
     _add_custom_target(${_component})
   ELSE()
     STRING(TOUPPER ${_component} _componentuppercase)
+
+    SET(_first_message "Error: Could not ${_description_string} disabled component ${_component}.")
+    DECORATE_WITH_STARS(${_first_message} _decorated_first_message)
+    SET(_second_message "Please reconfigure with -DDEAL_II_COMPONENT_${_componentuppercase}=ON")
+    DECORATE_WITH_STARS(${_second_message} _decorated_second_message)
+
     ADD_CUSTOM_TARGET(${_component}
       COMMAND
            ${CMAKE_COMMAND} -E echo ''
         && ${CMAKE_COMMAND} -E echo ''
         && ${CMAKE_COMMAND} -E echo '***************************************************************************'
```

Comment:
for consistency i would wrap line-of-starts in the same macro with
```
IF(${_message_length} EQ 0)
```
so that there is only one place where the length is defined.